pkg/datapath/loader/base.go

```diff
@@ -238,8 +238,7 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 		}
 
 		args[initArgMode] = mode
-		if option.Config.EnableNodePort &&
-			strings.ToLower(option.Config.Tunnel) != "disabled" {
+		if strings.ToLower(option.Config.Tunnel) != "disabled" {
```

Comment:
Nit (unrelated to your PR, but while we are here...)
```suggestion
		if option.Config.Tunnel != option.TunnelDisabled {
```

Reply:
Fixed. I simplified the code a bit and got rid of the `mode` variable while at it.